internal/pkg/api/handleCheckin_test.go

```diff
@@ -127,3 +136,41 @@ func TestFilterActions(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveSeqNo(t *testing.T) {
+	tests := []struct {
+		name  string
+		req   CheckinRequest
+		agent *model.Agent
+		resp  sqn.SeqNo
+	}{{
+		name: "empty ackToken",
+		req: CheckinRequest{
+			AckToken: new(string),
+		},
+		agent: &model.Agent{
+			ActionSeqNo: []int64{sqn.UndefinedSeqNo},
+		},
+		resp: []int64{sqn.UndefinedSeqNo},
+	}}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// setup mock CheckinT
+			logger := testlog.SetLogger(t)
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I took most of the setup code from `server_test.go`, might be simplified for a unit test